src/main/kotlin/io/github/corbym/dokker/dokker.kt

```diff
@@ -9,9 +10,45 @@ fun dokker(init: DokkerContainerBuilder.() -> Unit): DokkerContainer {
     return DokkerContainerBuilder().apply(init).build()
 }
 
+// Search order is
+// 1. process name set in code (if set, DokkerAutoProcessSearchResult.processName is not used)
+// 2. environment variable (if found in path)
+// 3. hardcoded "docker" (if found in path)
+// 3. hardcoded "podman" (if found in path)
+internal object DokkerAutoProcessSearchResult {
+
+    // Note that "command -v" - this is best-effort as adding an 'alias' will cause issues
+    // MacOS defaults to zsh shell.
+    //   Intellij does not read ~/.zprofile nor ~/.zshrc scripts so set environment variables in ~/.profile
+    //   Otherwise, "command -v" may not find the executable
+    private fun processFullPath(process: String): String? {
+        return try {
+            "command -v $process".runCommand()
+        } catch (e: Exception) {
```

Comment:
Can we remove catching the exception please? The internals to runCommand should deal with the process failing and printing out the correct response. Any unexpected exception should be propagated to the terminal anyway, if something else goes wrong.

Reply:
Oh, actually, we need that behaviour though as we're trying to see which one of the options is actually available.
`runCommand()` throws an `IllegalStateException` but we absolutely need to swallow the exception in order to continue the "search".

Reply:
There are a tonne of exceptions that could happen that we don't want to suppress if this command fails that have nothing to do with executing the command, er, `command` so I'd rather not catch everything. 

You can pass `fail=false` into runCommand and it shouldn't fail the command if Process fails to run it. Unfortunately, runCommand returns an empty string when it suppresses a fail (which it probably shouldn't 🫠), so we need to check ifBlank: 

```
"command -v $process".runCommand(fail = false).ifBlank { null }
```

Reply:
Sure. The intention was to ignore any actions taken by the search logic until a valid result is found, but happy to update as requested.

Reply:
thats actually another internal thing... we are not command execution library, so probably shouldnt expose `runCommand` but it will require much more work to internalize that...